src/plots/plots.js

```diff
@@ -854,7 +854,7 @@ plots.autoMargin = function(gd,id,o) {
     if(fullLayout.margin.autoexpand!==false) {
         if(!o) delete fullLayout._pushmargin[id];
         else {
-            var pad = o.pad||12;
+            var pad = o.pad === undefined ? 12 : o.pad;
```

Comment:
If the pad was set to `0`, it would cast to `false` and set `pad = 12`. I believe this was in error - it should be possible to set the pad to 0.